StortSpelProjekt/Engine/src/Renderer/Model.cpp

```diff
@@ -75,6 +75,22 @@ const SlotInfo* Model::GetSlotInfoAt(unsigned int index) const
 	return &m_SlotInfos[index];
 }
 
+void Model::UpdateSlotInfo()
+{
+	for (unsigned int i = 0; i < m_Size; i++)
+	{
+		m_SlotInfos[i] =
+		{
+		m_Meshes[i]->m_pSRV->GetDescriptorHeapIndex(),
+		m_Materials[i]->GetTexture(TEXTURE2D_TYPE::ALBEDO)->GetDescriptorHeapIndex(),
+		m_Materials[i]->GetTexture(TEXTURE2D_TYPE::ROUGHNESS)->GetDescriptorHeapIndex(),
+		m_Materials[i]->GetTexture(TEXTURE2D_TYPE::METALLIC)->GetDescriptorHeapIndex(),
+		m_Materials[i]->GetTexture(TEXTURE2D_TYPE::NORMAL)->GetDescriptorHeapIndex(),
+		m_Materials[i]->GetTexture(TEXTURE2D_TYPE::EMISSIVE)->GetDescriptorHeapIndex()
+		};
+	}
+}
```

Comment:
Is this meant to be called after a game-developer (for example) changes a texture on a model? If it isn't, isn't it better to just change the slotInfo automaticly when a new texture is set?

Reply:
Do not know. Just thought this would be written at a couple of places in the code and it deserved a function. Don't know if it is something that the game-devs should be calling. I'll set it to private and call it on setModel/Mesh/Material